src/ApplicationConfigIterator.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace TomPHP\ConfigServiceProvider;
+
+use RecursiveArrayIterator;
+use RecursiveIteratorIterator;
+
+final class ApplicationConfigIterator extends RecursiveIteratorIterator
+{
+    /**
+     * @var string[]
+     */
+    private $path = [];
+
+    /**
+     * @var string
+     */
+    private $separator;
+
+    /**
+     * @param ApplicationConfig $config
+     */
```

Comment:
@tomphp 

How about dropping the docblock?

As the argument is sufficiently type-hinted, the docblock doesn't really add any value, does it? Unless, of course, you plan to add a description here or something.